src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -333,88 +445,246 @@ private static void handleException(Exception e, HttpServletResponse resp) throw
+		if (StringUtils.isEmpty(uri) || "/".equals(uri)) {
+			return RequestType.SC_CONSOLE;
+		}
+		if (STS_TOKEN_URI.equals(uri)) {
+			return RequestType.STS_TOKEN;
+		}
+		if (ID_TOKEN_URI.equals(uri)) {
+			return RequestType.ID_TOKEN;
+		}
+		if (ACCESS_TOKEN_URI.equals(uri)) {
+			return RequestType.ACCESS_TOKEN;
+		}
+		throw new IllegalArgumentException("Unrecognized uri: "+uri);
+	}
+	
+	/**
+	 * Construct the HTTP redirect response that sends the browser to the Service Catalog console
+	 * @param claims
+	 * @param roleArn
+	 * @param selectedTeam
+	 * @param req
+	 * @param resp
+	 * @throws IOException
+	 */
+	void redirectToSCConsole(Claims claims, String roleArn, String selectedTeam, HttpServletRequest req, HttpServletResponse resp) throws IOException {
+		AssumeRoleRequest assumeRoleRequest = createAssumeRoleRequest(claims, roleArn, selectedTeam);
+		
+		AssumeRoleResult assumeRoleResult = stsClient.assumeRole(assumeRoleRequest);
+		Credentials credentials = assumeRoleResult.getCredentials();
+		// redirect to AWS login
+		String redirectURL = getConsoleLoginURL(req, credentials);
+		
+		resp.setHeader(LOCATION, redirectURL);
+		resp.setStatus(303);		
+	}
+	
+	/**
+	 * Create the HTTP response to download a file containing the STS token
+	 * which will allow the bearer to assume the end-user role. The file is
+	 * in the format of an AWS CLI config file.
+	 * 
+	 * @param claims
+	 * @param roleArn
+	 * @param selectedTeam
+	 * @param resp
+	 * @throws IOException
+	 */
+	void returnStsToken(Claims claims, String roleArn, String selectedTeam, HttpServletResponse resp) throws IOException {
+		AssumeRoleRequest assumeRoleRequest = createAssumeRoleRequest(claims, roleArn, selectedTeam);
+		
+		AssumeRoleResult assumeRoleResult = stsClient.assumeRole(assumeRoleRequest);
+		Credentials credentials = assumeRoleResult.getCredentials();
+		Map<String,String> sts = new HashMap<String,String>();
+		sts.put("AccessKeyId", credentials.getAccessKeyId());
+		sts.put("SecretAccessKey", credentials.getSecretAccessKey());
+		sts.put("SessionToken", credentials.getSessionToken());
+		
+		writeFileToResponse(createSerializedJSON(sts), "application/json", STS_TOKEN_FILE_NAME, resp);
+	}
+	
+	/**
+	 * Create serialized JSON for the given map of key/value pairs
+	 * @param content
+	 * @return
+	 */
+	public static String createSerializedJSON(Map<String,String> content) {
+		JSONObject o = new JSONObject();
+		for (Map.Entry<String,String> entry : content.entrySet()) {
+			o.put(entry.getKey(), entry.getValue());
+		}
+		return o.toString();
+	}
+	
+	/**
+	 * Create the HTTP response for downloading a file
+	 * 
+	 * @param content the file content
+	 * @param filename the file name
+	 * @param resp the HTTP response to write the result to
+	 * @throws IOException
+	 */
+	public static void writeFileToResponse(String content, String contentType, String filename, HttpServletResponse resp) throws IOException {
```

Comment:
The difference you suggest would just be a change to two response headers (the response body is the same).  Why would such a change make it "easier to integrate with AWS CLI"?

Reply:
As the [sourcing credentials with external process feature](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html) doc suggest..
```
credential_process = "/Users/Dave/path/to/credentials.sh" parameterWithoutSpaces "parameter with spaces"
```

I envision the credentials.sh file containing just a simple `curl ${STS_TOKEN_ENDPOINT}` to get the credentials needed for the AWS CLI.  if the login app only offers a file we would need to do extra work to extract the credentials from the file.

Also if we really want the credentials in a file we can just pass the `-O` flag to the same command,  `curl -O my_sts_creds.txt ${STS_TOKEN_ENDPOINT}`

Reply:
> we would need to do extra work to extract the credentials from the file.
I don't think this is correct.  I believe the response headers are directives to the browser and ignored by cUrl.  If you have evidence to the contrary, please share, otherwise I suggest you approve this PR and try using cUrl.